### PR TITLE
DON'T MERGE: Services pt 2

### PR DESCRIPTION
**Pull request by deanrobertcook**
#### APK
[Download build #11445](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11445/artifact/build/artifact/wire-dev-PR1722-11445.apk)
[Download build #11448](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11448/artifact/build/artifact/wire-dev-PR1722-11448.apk)
[Download build #11449](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11449/artifact/build/artifact/wire-dev-PR1722-11449.apk)
[Download build #11450](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11450/artifact/build/artifact/wire-dev-PR1722-11450.apk)
[Download build #11454](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11454/artifact/build/artifact/wire-dev-PR1722-11454.apk)
[Download build #11459](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11459/artifact/build/artifact/wire-dev-PR1722-11459.apk)
[Download build #11460](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11460/artifact/build/artifact/wire-dev-PR1722-11460.apk)
[Download build #11461](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11461/artifact/build/artifact/wire-dev-PR1722-11461.apk)
[Download build #11462](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11462/artifact/build/artifact/wire-dev-PR1722-11462.apk)
[Download build #11464](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11464/artifact/build/artifact/wire-dev-PR1722-11464.apk)
[Download build #11467](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11467/artifact/build/artifact/wire-dev-PR1722-11467.apk)
[Download build #11468](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11468/artifact/build/artifact/wire-dev-PR1722-11468.apk)
[Download build #11470](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11470/artifact/build/artifact/wire-dev-PR1722-11470.apk)
[Download build #11472](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11472/artifact/build/artifact/wire-dev-PR1722-11472.apk)
[Download build #11474](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11474/artifact/build/artifact/wire-dev-PR1722-11474.apk)
[Download build #11475](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11475/artifact/build/artifact/wire-dev-PR1722-11475.apk)
[Download build #11476](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11476/artifact/build/artifact/wire-dev-PR1722-11476.apk)
[Download build #11477](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11477/artifact/build/artifact/wire-dev-PR1722-11477.apk)
[Download build #11479](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11479/artifact/build/artifact/wire-dev-PR1722-11479.apk)
[Download build #11483](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11483/artifact/build/artifact/wire-dev-PR1722-11483.apk)
[Download build #11486](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11486/artifact/build/artifact/wire-dev-PR1722-11486.apk)
[Download build #11488](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11488/artifact/build/artifact/wire-dev-PR1722-11488.apk)
[Download build #11489](http://192.168.120.33:8080/job/Pull%20Request%20Builder/11489/artifact/build/artifact/wire-dev-PR1722-11489.apk)

### CODE REVIEW

**Comment by someoneEsle** [comment]
- [ ] Bug 1
Right below the Manage Services button I see a "SERVICES" label which is not in the specs

**Comment by eike96** [comment]
Is this in the latest version of experimental? I dont see the "manage services" button and i am a team owner

**Comment by someoneEsle** [comment]
@eike96 it's a bug, you can see the button if you log in as team admin though for now

File: app/src/main/scala/com/waz/zclient/common/views/ImageAssetDrawable.scala [hunk 1]
```diff
@@ -283,45 +283,50 @@ class IntegrationAssetDrawable (
...
-    tempCanvas.drawBitmap(bm, matrix, null)
+    val strokeW = strokeWidth(bounds.width)
 
+    borderPaint.setStrokeWidth(strokeW)
+    val outerRect = new RectF(strokeW, strokeW, bounds.width - strokeW, bounds.height - strokeW)
+    val backgroundRect = new RectF(strokeW, strokeW, bounds.width - strokeW, bounds.height - strokeW)
+    val innerRect = new RectF(padding * bounds.width, padding * bounds.height, bounds.width - padding * bounds.width, bounds.height - padding * bounds.height)
+
+    val matrix2 = new Matrix()
+    scaleType(matrix2, bm.getWidth, bm.getHeight, Dim2(innerRect.width.toInt, innerRect.height.toInt))
```
Comment (makingthematrix) [inline app/src/main/scala/com/waz/zclient/common/views/ImageAssetDrawable.scala]: Ok, I understand how it works - we use the case object's `apply` method in order to modify `matrix2` - but this is very confusing piece of code. It'd be much more straightforward to have a method which would return an already initialized matrix.

**Pull request merged**